Cherrypick precompiles fixes

nanocryk opened the pull request:
upstream PR: https://github.com/paritytech/frontier/pull/961

librelois commented:
@nanocryk can you add some rust unit tests that check the good behavior of these precompiles with some hardcoded inputs?

nanocryk commented:
Added tests for ecrecover. The other 2 changes already have tests (changed an existing test that was non-compliant for modexp padding, and added a new test with hardcoded input for modexp gas used)
